### PR TITLE
Provide embedded documentation

### LINKED ISSUE #35: Provide embedded documentation
It should be nice to have some documentation about:

 * the kafka settings
 * the Kafka explorer

Providing an embedded documentation gives the capability to read the documentation with webrowser (by reading pages from the github repository) or inside vscode.

We did that for vscode-xml, the documentation https://github.com/redhat-developer/vscode-xml/blob/master/docs/README.md can be read too inside vscode with the vscode `Open XML Documentation `command https://github.com/redhat-developer/vscode-xml/blob/master/docs/Commands.md#commands

### PR DESCRIPTION
**Pull request by angelozerr**
Provide embedded documentation

Fixes #35

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR provides 

 * online vscode-kafka documentation
 
image

 * offline vscode-documentation . 
 
KakaEmbedDocDemo

The content of the doc must be improved a lot but the mechanism is done.  The following idea is to bind documentation from the settings to the embed doc.

**Comment by fbricon** [comment]
we need to update the images matching the more recent changes (syntax highlighing makes things look much better)

**Comment by angelozerr** [comment]
> we need to update the images matching the more recent changes (syntax highlighing makes things look much better)

I agree but the goal of this PR is to initialize embed documentation mechanism. We could do that in an another PR I think, because IMHO I think documentation should be more structured.

File: package.json [hunk 1]
```diff
@@ -23,6 +23,8 @@
         "kafka"
```
Comment (fbricon) [inline package.json:23]: please add links from kafka.producers.fakerjs.* settings to the producing doc

File: README.md [hunk 2]
```diff
@@ -20,12 +20,17 @@ Planned features in no particular order:
 - More administration features
 - Update configs
 
+You might also find useful information in the [Online VScode Kafka Documentation](https://github.com/jlandersen/vscode-kafka/blob/master/docs/README.md)
+or you can read this documentation inside vscode with the command `Open VScode Kafka Documentation` available with `Ctrl+Shift+P`:
```
Comment (fbricon) [inline README.md:24]: inside vscode => inside your editor

can run in vscodium, che, theia...

File: package.json [hunk 3]
```diff
@@ -174,6 +176,11 @@
             }
         ],
         "commands": [
+            {
+                "command": "vscode-kafka.open.docs.home",
+                "title": "Open VSCode Kafka Documentation",
```
Comment (fbricon) [inline package.json:181]: There's already a Kafka prefix:
Screenshot 2021-01-25 at 14 29 59

"Open Documentation" would be better

File: package.json (hunk 1 above)
Comment (angelozerr) [reply package.json:23]: fixed for all settings. The description is not very helpful for the moment, but in the future, we could improve it with screenshots to see in action the settings result.

File: README.md (hunk 2 above)
Comment (angelozerr) [reply README.md]: fixed

**Comment by angelozerr** [comment]
Here a demo with embed doc. Each settings from package.json are linked to the doc.

EmbedDoc

File: README.md [hunk 4]
```diff
@@ -20,12 +20,18 @@ Planned features in no particular order:
 - More administration features
 - Update configs
 
+You might also find useful information in the [Online Documentation](https://github.com/jlandersen/vscode-kafka/blob/master/docs/README.md)
+or you can read this documentation inside your editor (can run in vscode, vscodium, che, theia...) with the command `Open Documentation` available with `Ctrl+Shift+P`:
```
Comment (fbricon) [inline README.md:24]: remove `(can run in vscode, vscodium, che, theia...)`, it was meant as a review  comment  ;-)

File: docs/Producing.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+topic: json-events
+{
+    "type": "my_test_event"
+}
+```
+
+For actually producing a single record, click on the "Produce record" link above the PRODUCER line; for producing 10 records, click on "Produce record x 10".
+The log about produced messages is printed in the "Kafka Producer Log" Output view.
+
+# Faker.js
```
Comment (fbricon) [inline docs/Producing.md:23]: `# Randomized content`

Comment (angelozerr) [reply docs/Producing.md]: fixed

File: README.md (hunk 4 above)
Comment (angelozerr) [reply README.md]: fixed

File: docs/Consuming.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+Consuming topics can be done by right clicking a topic in the explorer or from the command palette.  Some things to note about consuming:
+
+* UTF-8 encoded keys and values only. If data is encoded differently, it will not be pretty.
+* One consumer group is created per topic (may change in the future to just have one for the extension).
+
+Consumers are based on virtual documents, available in the VS Code extension API. A consumer will keep running even if you close the document in the editor. You should make sure to close the consumer explicitly, either via the command palette, the status bar element or the start/stop action button as well. The VS Code API does not support detecting if a virtual document is closed immediately. Instead, the underlying virtual document is automatically closed after two minutes if the document is closed in the editor.
+
+# Preferences
+
+## kafka.consumers.offset
```
Comment (fbricon) [inline docs/Consuming.md:12]: To make the preference keys more readable, I found that using this pattern works better:

``` 
### `kafka.consumers.offset`
```

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# VSCode Kafka Documentation
+
+Welcome to the [vscode-kafka](https://github.com/jlandersen/vscode-kafka) documentation.
+
+* [Kafka Explorer](Explorer.md#explorer)
+* [Producing](Producing.md#producing)
```
Comment (fbricon) [inline docs/README.md:6]: Producing messages

File: docs/README.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# VSCode Kafka Documentation
+
+Welcome to the [vscode-kafka](https://github.com/jlandersen/vscode-kafka) documentation.
+
+* [Kafka Explorer](Explorer.md#explorer)
+* [Producing](Producing.md#producing)
+* [Consuming](Consuming.md#consuming)
```
Comment (fbricon) [inline docs/README.md:7]: Consuming messages

File: docs/Consuming.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# Consuming
```
Comment (fbricon) [inline docs/Consuming.md:1]: Consuming messages

File: docs/Producing.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
+# Producing
```
Comment (fbricon) [inline docs/Producing.md:1]: Producing messages

File: docs/Producing.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
+# Producing
+
+Producing can be done by creating a `.kafka` file. Write simple producers using the following format:
```
Comment (fbricon) [inline docs/Producing.md:3]: Producing messages

File: docs/Consuming.md (hunk 9 above)
Comment (angelozerr) [reply docs/Consuming.md:1]: Is not consuming topics?

File: README.md [hunk 12]
```diff
@@ -20,14 +20,25 @@ Planned features in no particular order:
...
+You might also find useful information in the [Online Documentation](https://github.com/jlandersen/vscode-kafka/blob/master/docs/README.md)
+or you can read this documentation inside your editor with the command `Open Documentation` available with `Ctrl+Shift+P`:
 
-![Screenshot-2](assets/screen-2.png)
+![Open Documentation](docs/assets/open-doc-cmd.png)
 
-![Screenshot-3](assets/screen-3.png)
+## Kafka explorer
+
+Kafka explorer show configured clusters with their topics, brokers,consumers and configurations.
```
Comment (angelozerr) [inline README.md:30]: Is it OK for you @fbricon ?

File: docs/Producing.md (hunk 10 above)
Comment (angelozerr) [reply docs/Producing.md:1]: fixed

File: docs/Producing.md (hunk 11 above)
Comment (angelozerr) [reply docs/Producing.md:3]: fixed

File: docs/Consuming.md (hunk 6 above)
Comment (angelozerr) [reply docs/Consuming.md]: fixed

File: docs/README.md (hunk 8 above)
Comment (angelozerr) [reply docs/README.md]: I have fixed that but is not Consuming topics?

File: docs/Producing.md (hunk 10 above)
Comment (angelozerr) [reply docs/Producing.md:1]: fixed

File: docs/Producing.md (hunk 11 above)
Comment (angelozerr) [reply docs/Producing.md:3]: fixed

File: README.md (hunk 12 above)
Comment (fbricon) [reply README.md:30]: > The Kafka explorer shows configured clusters with their topics, brokers, consumers and configurations.

File: docs/README.md (hunk 8 above)
Comment (fbricon) [reply docs/README.md]: both are probably correct. Consuming messages should be understood by most people though.

File: docs/README.md (hunk 7 above)
Comment (angelozerr) [reply docs/README.md]: fixed

File: docs/README.md (hunk 8 above)
Comment (angelozerr) [reply docs/README.md]: fixed

File: README.md (hunk 12 above)
Comment (angelozerr) [reply README.md]: fixed

File: docs/Consuming.md (hunk 9 above)
Comment (angelozerr) [reply docs/Consuming.md]: fixed

**Comment by fbricon** [comment]
Thanks @angelozerr. We still need to improve the doc but at least we now have the foundation to do it.

**Pull request merged**